### PR TITLE
[Doc] add col mode partial updates

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/loading/Load_to_Primary_Key_tables.md [hunk 1]
```diff
@@ -463,12 +463,14 @@ SELECT * FROM table3;
 
 ## 部分更新
 
-自 StarRocks v2.2 起，主键表支持部分更新 (Partial Update)，您可以选择只更新部分指定的列。这里以 CSV 格式的数据文件为例进行说明。
+主键模型还支持部分列更新（Partial Updates），并且针对不同的数据更新场景，提供了行模式和列模式两种部分列更新，在不影响查询性能的同时，尽可能地降低部分列更新的开销，从而能够保证更新的实时性。行模式比较适用于小批量的实时更新场景，列模式适用于大批量的批处理更新场景。
```
Comment (jaogoy) [inline docs/zh/loading/Load_to_Primary_Key_tables.md:466]: “列模式适用于大批量的批处理更新场景。”
列模式适用于大批量的**少量列**的批处理更新场景。

File: docs/zh/loading/Flink-connector-starrocks.md [hunk 2]
```diff
@@ -117,6 +117,8 @@ Flink connector JAR 文件的命名格式如下：
 | sink.properties.column_separator  | No       | \t            | CSV 数据的列分隔符。                                         |
 | sink.properties.row_delimiter     | No       | \n            | CSV 数据的行分隔符。                                         |
 | sink.properties.max_filter_ratio  | No       | 0             | 导入作业的最大容错率，即导入作业能够容忍的因数据质量不合格而过滤掉的数据行所占的最大比例。取值范围：0~1。默认值：0 。详细信息，请参见  [STREAM LOAD](../sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md)。 |
+| sink.properties.partial_update      | 否      | `FALSE` | 是否使用部分列更新。取值包括 `TRUE` 和 `FALSE`。默认值：`FALSE`。                                                                                                                                                                                             |
+| sink.properties.partial_update_mode | 否      | `auto` | 指定部分更新的模式，取值包括 `auto` 和 `column`。<ul><li>`auto`（默认值），表示由系统通过分析更新语句以及其涉及的列，自动判断执行部分更新时使用的模式。如果满足如下标准，则系统自动使用列模式：</li><ul><li>更新的列数占所有列数的百分比小于 30%，并且更新的列数少于 4 个。</li><li>更新语句中没有使用 WHERE 条件。反之，则系统不会使用列模式。</li></ul><li>`column`，指定使用列模式执行部分更新，比较适用于涉及少数列并且大量行的部分列更新场景。在该场景，开启列模式，更新速度更快。例如，在一个包含 100 列的表中，每次更新 10 列（占比 10%）并更新所有行，则开启列模式，更新性能将提高 10 倍。</li></ul>   |
```
Comment (luohaha) [inline docs/zh/loading/Flink-connector-starrocks.md:121]: 目前只有`update`语句会做auto推断，其他的connector，以及stream load/broker load都是只有row/column两种模式选择，默认是row模式。

File: docs/zh/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md [hunk 3]
```diff
@@ -116,6 +116,8 @@ PROPERTIES ("<key1>" = "<value1>"[, "<key2>" = "<value2>" ...])
 | strict_mode               | 否           | 是否开启严格模式。取值范围：`TRUE` 或者 `FALSE`。默认值：`FALSE`。开启后，如果源数据某列的值为 `NULL`，但是目标表中该列不允许为 `NULL`，则该行数据会被过滤掉。<br />关于该模式的介绍，参见[严格模式](../../../loading/load_concept/strict_mode.md)。|
 | log_rejected_record_num | 否 | 指定最多允许记录多少条因数据质量不合格而过滤掉的数据行数。该参数自 3.1 版本起支持。取值范围：`0`、`-1`、大于 0 的正整数。默认值：`0`。<ul><li>取值为 `0` 表示不记录过滤掉的数据行。</li><li>取值为 `-1` 表示记录所有过滤掉的数据行。</li><li>取值为大于 0 的正整数（比如 `n`）表示每个 BE 节点上最多可以记录 `n` 条过滤掉的数据行。</li></ul> |
 | timezone                  | 否           | 该参数的取值会影响所有导入涉及的、跟时区设置有关的函数所返回的结果。受时区影响的函数有 strftime、alignment_timestamp 和 from_unixtime 等，具体请参见[设置时区](../../../administration/management/timezone.md)。导入参数 `timezone` 设置的时区对应[设置时区](../../../administration/management/timezone.md)中所述的会话级时区。 |
+| partial_update | 否 |是否使用部分列更新。取值包括 `TRUE` 和 `FALSE`。默认值：`FALSE`。|
+| partial_update_mode | 否 | 指定部分更新的模式，取值包括 `auto` 和 `column`。<ul><li>`auto`（默认值），表示由系统通过分析更新语句以及其涉及的列，自动判断执行部分更新时使用的模式。如果满足如下标准，则系统自动使用列模式：</li><ul><li>更新的列数占所有列数的百分比小于 30%，并且更新的列数少于 4 个。</li><li>更新语句中没有使用 WHERE 条件。反之，则系统不会使用列模式。</li></ul><li>`column`，指定使用列模式执行部分更新，比较适用于涉及少数列并且大量行的部分列更新场景。在该场景，开启列模式，更新速度更快。例如，在一个包含 100 列的表中，每次更新 10 列（占比 10%）并更新所有行，则开启列模式，更新性能将提高 10 倍。</li></ul>|
```
Comment (luohaha) [inline docs/zh/sql-reference/sql-statements/data-manipulation/CREATE_ROUTINE_LOAD.md:120]: routine load不支持partial_update_mode 设置，默认只有row模式。

File: docs/zh/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md [hunk 4]
```diff
@@ -143,6 +143,8 @@ http://<fe_host>:<fe_http_port>/api/<database_name>/<table_name>/_stream_load
 -H "strict_mode: true | false"
 -H "timezone: <string>"
 -H "load_mem_limit: <num>"
+-H "partial_update: true | false"
+-H "partial_update_mode: auto | column"
```
Comment (luohaha) [inline docs/zh/sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md:147]: -H "partial_update_mode: row | column"

File: docs/zh/loading/Kafka-connector-starrocks.md [hunk 5]
```diff
@@ -212,6 +212,8 @@ MySQL [example_db]> select * from test_tbl;
...
 | sink.properties.format              | 否       | json                                                         | Stream Load 导入时的数据格式。取值为 CSV 或者 JSON。默认为JSON。更多参数说明，参见 [CSV 适用参数](../sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md#csv-适用参数)和 [JSON 适用参数](../sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md#json-适用参数)。 |
+| sink.properties.partial_update      | 否      | `FALSE` | 是否使用部分列更新。取值包括 `TRUE` 和 `FALSE`。默认值：`FALSE`。                                                                                                                                                                                             |
+| sink.properties.partial_update_mode | 否      | `auto` | 指定部分更新的模式，取值包括 `auto` 和 `column`。<ul><li>`auto`（默认值），表示由系统通过分析更新语句以及其涉及的列，自动判断执行部分更新时使用的模式。如果满足如下标准，则系统自动使用列模式：</li><ul><li>更新的列数占所有列数的百分比小于 30%，并且更新的列数少于 4 个。</li><li>更新语句中没有使用 WHERE 条件。反之，则系统不会使用列模式。</li></ul><li>`column`，指定使用列模式执行部分更新，比较适用于涉及少数列并且大量行的部分列更新场景。在该场景，开启列模式，更新速度更快。例如，在一个包含 100 列的表中，每次更新 10 列（占比 10%）并更新所有行，则开启列模式，更新性能将提高 10 倍。</li></ul>   |
```
Comment (luohaha) [inline docs/zh/loading/Kafka-connector-starrocks.md:216]: 这个也是同上的，只有row和column，这个文档里提到的接口都只有row和column

File: docs/en/loading/Load_to_Primary_Key_tables.md [hunk 6]
```diff
@@ -465,12 +465,14 @@ As shown in the preceding query result, the record whose `id` is `101` in `examp
 
 ## Partial updates
 
-Since v2.2, StarRocks supports updating only the specified columns of a Primary Key table. This section uses CSV as an example to describe how to perform partial updates.
+Primary Key tables also support partial updates, and provide two modes, row mode and column mode, for different data updates scenarios. These two modes of partial updates can minimize the overhead of partial updates as much as possible while guaranteeing query performance, ensuring real-time updates. Row mode is more suitable for real-time update scenarios involving many columns and small batches. Column mode is suitable for batch processing update scenarios involving a few columns and a large number of rows.
```
Comment (EsoragotoSpirit) [inline docs/en/loading/Load_to_Primary_Key_tables.md:468]: for different data update scenarios

File: docs/en/sql-reference/sql-statements/data-manipulation/BROKER_LOAD.md [hunk 7]
```diff
@@ -611,6 +611,15 @@ The following parameters are supported:
...
   StarRocks allows setting the `priority` parameter for a Broker Load job since v2.5.
 
+- `partial_update`
+
+  Whether to use partial updates. Valid values: `TRUE` and `FALSE`. Default value: `FALSE`.
+
+
+- `partial_update_mode`
+
+  Specifies the mode for partial updates. Valid values: `row` and `column`. <ul><li> The value `row` (default) means partial updates in row mode, which is more suitable for real-time updates with many columns and small batches.</li><li>The value `column` means partial updates in column mode, which is more suitable for batch updates with few columns and many rows. In such scenarios, enabling the column mode offers faster update speeds. For example, in a table with 100 columns, if only 10 columns (10% of the total) are updated for all rows, the update speed of the column mode is 10 times faster.</li></ul> |
```
Comment (EsoragotoSpirit) [inline docs/en/sql-reference/sql-statements/data-manipulation/BROKER_LOAD.md:621]: 最后的 pipe 多余

File: docs/en/loading/Flink-connector-starrocks.md [hunk 8]
```diff
@@ -227,11 +227,17 @@ In your Maven project's `pom.xml` file, add the Flink connector as a dependency
 **Default value**: 0<br/>
 **Description**: The maximum error tolerance of the Stream Load. It's the maximum percentage of data records that can be filtered out due to inadequate data quality. Valid values: `0` to `1`. Default value: `0`. See [Stream Load](../sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md) for details.
 
-### sink.parallelism
```
Comment (EsoragotoSpirit) [inline docs/en/loading/Flink-connector-starrocks.md:230]: 这个参数中文文档里没有删除？

File: docs/en/loading/Flink-connector-starrocks.md [hunk 9]
```diff
@@ -227,11 +227,17 @@ In your Maven project's `pom.xml` file, add the Flink connector as a dependency
...
+### starrocks.write.properties.partial_update
 
-**Required**: No<br/>
-**Default value**: NONE<br/>
-**Description**: The parallelism of the connector. Only available for Flink SQL. If not set, Flink planner will decide the parallelism. In the scenario of multi-parallelism, users need to guarantee data is written in the correct order.
+**Required**:  NO<br/>
```
Comment (EsoragotoSpirit) [inline docs/en/loading/Flink-connector-starrocks.md:232]: extra space

File: docs/en/loading/Flink-connector-starrocks.md [hunk 10]
```diff
@@ -227,11 +227,17 @@ In your Maven project's `pom.xml` file, add the Flink connector as a dependency
...
+**Default value**: `FALSE`<br/>
+**Description**: Whether to use partial updates. Valid values: `TRUE` and `FALSE`. Default value: `FALSE`.
```
Comment (EsoragotoSpirit) [inline docs/en/loading/Flink-connector-starrocks.md:234]: Default value: `FALSE`, indicating to disable this feature.

Comment (EsoragotoSpirit) [reply docs/en/loading/Flink-connector-starrocks.md:234]: 下同

File: docs/en/loading/Flink-connector-starrocks.md (hunk 8 above)
Comment (hellolilyliuyi) [reply docs/en/loading/Flink-connector-starrocks.md:230]: 英文中出现了两次 删除多余的

**Review by EsoragotoSpirit — DISMISSED** [review]

File: docs/en/loading/Flink-connector-starrocks.md [hunk 11]
```diff
@@ -227,11 +227,17 @@ In your Maven project's `pom.xml` file, add the Flink connector as a dependency
 **Default value**: 0<br/>
 **Description**: The maximum error tolerance of the Stream Load. It's the maximum percentage of data records that can be filtered out due to inadequate data quality. Valid values: `0` to `1`. Default value: `0`. See [Stream Load](../sql-reference/sql-statements/data-manipulation/STREAM_LOAD.md) for details.
 
-### sink.parallelism
+### starrocks.write.properties.partial_update
```
Comment (banmoy) [inline docs/en/loading/Flink-connector-starrocks.md:230]: It should be `sink.properties.xxxx`

File: docs/en/loading/Flink-connector-starrocks.md [hunk 12]
```diff
@@ -227,11 +227,17 @@ In your Maven project's `pom.xml` file, add the Flink connector as a dependency
...
 
-**Required**: No<br/>
-**Default value**: NONE<br/>
-**Description**: The parallelism of the connector. Only available for Flink SQL. If not set, Flink planner will decide the parallelism. In the scenario of multi-parallelism, users need to guarantee data is written in the correct order.
+**Required**: NO<br/>
+**Default value**: `FALSE`<br/>
+**Description**: Whether to use partial updates. Valid values: `TRUE` and `FALSE`. Default value: `FALSE`, indicating to disable this feature.
+
+### starrocks.write.properties.partial_update_mode
```
Comment (banmoy) [inline docs/en/loading/Flink-connector-starrocks.md:236]: ditto

**Review by banmoy — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Pull request merged**